hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/ws/command/ForwardCommand.java

```diff
@@ -26,37 +23,48 @@
 import org.hawkular.bus.common.MessageId;
 import org.hawkular.bus.common.MessageProcessor;
 import org.hawkular.bus.common.producer.ProducerConnectionContext;
-import org.hawkular.cmdgw.api.DeployApplicationRequest;
+import org.hawkular.cmdgw.api.AuthMessage;
+import org.hawkular.cmdgw.api.AuthMessage.RecipientType;
 import org.hawkular.cmdgw.api.GenericSuccessResponse;
 import org.hawkular.cmdgw.ws.Constants;
 import org.hawkular.cmdgw.ws.MsgLogger;
-import org.hawkular.cmdgw.ws.command.Command;
-import org.hawkular.cmdgw.ws.command.CommandContext;
-import org.hawkular.inventory.api.model.CanonicalPath;
 
 /**
  * UI client requesting to send a file to a remote feed.
  */
-public class DeployApplicationCommand implements Command<DeployApplicationRequest, GenericSuccessResponse> {
-    public static final Class<DeployApplicationRequest> REQUEST_CLASS = DeployApplicationRequest.class;
+public class ForwardCommand implements Command<AuthMessage, GenericSuccessResponse> {
 
     @Override
-    public BasicMessageWithExtraData<GenericSuccessResponse> execute(DeployApplicationRequest request,
-            BinaryData binaryData, CommandContext context) throws Exception {
+    public BasicMessageWithExtraData<GenericSuccessResponse> execute(AuthMessage request, BinaryData binaryData,
+            CommandContext context) throws Exception {
 
-        // determine what feed needs to be sent the message
-        CanonicalPath resourcePath = CanonicalPath.fromString(request.getResourcePath());
-        String feedId = resourcePath.ids().getFeedId();
+        RecipientType recipientType = request.getRecipientType();
```

Comment:
I don't see any reason why we can't depend on inventory. That was the reason why we did the split - so that we could. Also, in the future, we are going to need inventory anyway. I suspect in the future we are going to have to do group commands where the client will send us a "groupPath" (or "groupIdentifier" or whatever) and we are going to have to ask inventory what resources are in that group and from there determine what feeds manage those resources.

In addition, we need inventory API to get the feed from the resourcePath. Otherwise, we are putting the responsibility on the client to know what feed is managing the resource it wants to operate on, and this is not good to do. See my previous long comment on this. The client should never have to know anything about the management infrastructure (such as knowledge of what management agents are managing what resources). That's the job of the hawkular server - the client should only care about the IDs of the actual resources it wants to operate on.